nemo_curator/utils/distributed_utils.py

```diff
@@ -617,54 +617,72 @@ def write_to_disk(
 
     Args:
         df: A Dask DataFrame.
-        output_file_dir: The output file path.
+        output_path: The output file path.
         write_to_filename: Boolean representing whether to write the filename using the "filename" column.
         keep_filename_column: Boolean representing whether to keep or drop the "filename" column, if it exists.
         output_type: The type of output file to write. Can be "jsonl" or "parquet".
 
     """
-    if write_to_filename and "filename" not in df.columns:
+
+    # output_path is a file name
+    if isinstance(output_path, str) and output_path.endswith(".jsonl"):
+        if df.npartitions == 1:
+            _write_to_jsonl_or_parquet(df.compute(), output_path, output_type)
```

Comment:
Instead of `df.compute()` I wonder if `df.map_partitions(_write_to_jsonl).compute()` or other equivalent like `df.get_partition(0)...` might work in this case. It avoids the latency of pulling the dataframe from one of the worker processes to the client process and instead uses the worker to write out to a single file instead.

Reply:
Very nice, thank you. Updated.